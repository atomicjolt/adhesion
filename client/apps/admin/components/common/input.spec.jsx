import React from 'react';
import { shallow } from 'enzyme';
import Input from './input';

describe('input', () => {
  let result;
  let props;
  let changed;

  beforeEach(() => {
    changed = false;
    props = {
      inputProps: {
        id: 'IM AN ID',
        value: 'IM A VALUE',
        checked: true,
        name: 'the name',
        type: 'radio',
        onChange: () => { changed = true; },
      },
      className: 'imaclass',
      labelText: 'IMA LABEL',
    };
    result = shallow(<Input {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the onChange function', () => {
    expect(changed).toBeFalsy();
    result.find('input').simulate('change');
    expect(changed).toBeTruthy();
  });
});
