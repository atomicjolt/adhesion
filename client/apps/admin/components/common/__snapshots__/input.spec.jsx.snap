// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input matches the snapshot 1`] = `
<div
  className="c-input--container"
>
  <label
    className="imaclass"
    htmlFor="IM AN ID"
  >
    <input
      checked={true}
      id="IM AN ID"
      name="the name"
      onChange={[Function]}
      type="radio"
      value="IM A VALUE"
    />
    <span>
      IMA LABEL
    </span>
  </label>
</div>
`;
